v2/src/layer.ts

```diff
@@ -55,48 +55,36 @@ export function applyLayers(
     let extensionLayerArn;
     if (lambdaRuntimeType === RuntimeType.PYTHON) {
       if (pythonLayerVersion === undefined) {
-        const errorMessage = getMissingLayerVersionErrorMsg(lam.node.id, "Python", "python");
-        log.error(errorMessage);
-        errors.push(errorMessage);
-        return;
+        return handleLayerError(errors, lam.node.id, "Python", "python");
       }
-      lambdaLayerArn = getLambdaLayerArn(region, pythonLayerVersion, runtime, isARM, false, accountId);
+      lambdaLayerArn = getLambdaLayerArn(region, pythonLayerVersion, runtime, isARM, accountId);
       log.debug(`Using Python Lambda layer: ${lambdaLayerArn}`);
       addLayer(lambdaLayerArn, false, scope, lam, runtime);
     }
 
     if (lambdaRuntimeType === RuntimeType.NODE) {
       if (nodeLayerVersion === undefined) {
-        const errorMessage = getMissingLayerVersionErrorMsg(lam.node.id, "Node.js", "node");
-        log.error(errorMessage);
-        errors.push(errorMessage);
-        return;
+        return handleLayerError(errors, lam.node.id, "Node.js", "node");
```

Comment:
Using a method for error handling is a great idea, unfortunately this looks like the function is returning something, which never does.
It would be great to think about enforcing some typescript rule to avoid these cases